src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -101,6 +102,51 @@ case class NamespaceWithWorkflow(importedAs: Option[String],
     }
   }
 
+  private def declarationLookupFunction(decl: Declaration, inputs: Map[FullyQualifiedName, WdlValue]): String => WdlValue ={
+    def identifierLookup(string: String): WdlValue = {
+
+      /* This is a scope hierarchy to search for the variable `string`.  If `decl.scopeFqn` == "a.b.c"
+       * then `hierarchy` should be Seq("a.b.c", "a.b", "a")
+       */
+      val hierarchy = decl.scopeFqn.split("\\.").reverse.tails.toSeq.map {_.reverse.mkString(".")}
+
+      /* Attempt to resolve the string in each scope */
+      val attemptedValues = hierarchy.map {scope => inputs.get(s"$scope.$string")}
+      attemptedValues.flatten.headOption.getOrElse {
+        throw new WdlExpressionException(s"Could not find a value for $string")
+      }
+    }
+    identifierLookup
+  }
+
+  /* Some declarations need a value from the user and some have an expression attached to them.
+   * For the declarations that have an expression attached to it already, evaluate the expression
+   * and return the value for storage in the symbol store
+   */
+  def staticDeclarationsRecursive(userInputs: WorkflowCoercedInputs): Try[WorkflowCoercedInputs] = {
+    import scala.collection.mutable
+    val collected = mutable.Map[String, WdlValue]()
+    class NoFunctions extends WdlFunctions {
+      def getFunction(name: String): WdlFunction = throw new UnsupportedOperationException("No functions should be called in this test")
+    }
+
+    val allDeclarations = workflow.declarations ++ workflow.calls.flatMap {_.task.declarations}
+
+    val evaluatedDeclarations = allDeclarations.filter {_.expression.isDefined}.map {decl =>
+      val value = decl.expression.get.evaluate(declarationLookupFunction(decl, collected.toMap ++ userInputs), new NoFunctions)
+      collected += (decl.fullyQualifiedName -> value.get)
+      decl.fullyQualifiedName -> value
+    }.toMap
```

Comment:
This code finds all Declarations with a right-hand side expression, for example:

```
String x = "foo" + " bar"
String y = x + baz
```

Then, it evaluates the expressions (in order) and adds the each evaluated WdlValue to the mutable map so other expressions can utilize that value.  We want to support the `String y` case above.  The only other way I can think of to do this without a mutable map and only changing this function would be to do it in multiple passes

Really it'd be much better if this could read/write directly from the symbol store so it doesn't have to go through this at all, though that required much more refactoring